nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ExtractText.java

```diff
@@ -372,7 +411,23 @@ public void process(InputStream in) throws IOException {
                     String value = matcher.group(i);
                     if (value != null && !value.isEmpty()) {
                         if (value.length() > maxCaptureGroupLength) {
-                            value = value.substring(0, maxCaptureGroupLength);
+                            if (logger.isDebugEnabled()) {
+                                logger.debug(String.format(errorMessageOnCaptureGroupLengthExcess, value, maxCaptureGroupLength, flowFile));
+                            }
+                            switch (excessPolicy) {
+                                case TRUNCATE:
+                                    value = value.substring(0, maxCaptureGroupLength);
+                                    break;
+                                case FAIL:
+                                    final String errorMsg = String.format(errorMessageOnCaptureGroupLengthExcess,
+                                            value, maxCaptureGroupLength, flowFile);
+                                    flowFile = session.putAttribute(flowFile, ERROR_MSG_KEY, errorMsg);
+                                    session.getProvenanceReporter().modifyAttributes(flowFile);
+                                    session.transfer(flowFile, FAILURE);
+                                    logger.info("Transferred FlowFile {} to FAILURE relationship due to excess of capture group maximum length.",
```

Comment:
Would avoid capitalizing FAILURE here since the relationship name is failure in lowercase.